pkg/agent/controller/egress/egress_controller.go

```diff
@@ -0,0 +1,672 @@
+// Copyright 2021 Antrea Authors
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+package egress
+
+import (
+	"context"
+	"fmt"
+	"net"
+	"reflect"
+	"strings"
+	"sync"
+	"time"
+
+	"k8s.io/apimachinery/pkg/api/errors"
+	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+	"k8s.io/apimachinery/pkg/fields"
+	"k8s.io/apimachinery/pkg/util/sets"
+	"k8s.io/apimachinery/pkg/util/wait"
+	"k8s.io/apimachinery/pkg/watch"
+	"k8s.io/client-go/tools/cache"
+	"k8s.io/client-go/util/workqueue"
+	"k8s.io/klog"
+
+	"github.com/vmware-tanzu/antrea/pkg/agent"
+	"github.com/vmware-tanzu/antrea/pkg/agent/interfacestore"
+	"github.com/vmware-tanzu/antrea/pkg/agent/openflow"
+	"github.com/vmware-tanzu/antrea/pkg/agent/route"
+	cpv1b2 "github.com/vmware-tanzu/antrea/pkg/apis/controlplane/v1beta2"
+	crdv1a2 "github.com/vmware-tanzu/antrea/pkg/apis/crd/v1alpha2"
+	crdinformers "github.com/vmware-tanzu/antrea/pkg/client/informers/externalversions/crd/v1alpha2"
+	crdlisters "github.com/vmware-tanzu/antrea/pkg/client/listers/crd/v1alpha2"
+	"github.com/vmware-tanzu/antrea/pkg/k8s"
+)
+
+const (
+	controllerName = "AntreaAgentEgressController"
+	// How long to wait before retrying the processing of an Egress change.
+	minRetryDelay = 5 * time.Second
+	maxRetryDelay = 300 * time.Second
+	// Default number of workers processing an Egress change.
+	defaultWorkers = 4
+	// Disable resyncing.
+	resyncPeriod time.Duration = 0
+	// maxEgressIPsPerNode is the maximum number of Egress IPs can be configured on a Node.
+	maxEgressIPsPerNode = 256
```

Comment:
I think 0 is for default IP, so the max number should be 255?

Reply:
Thanks, will fix. And switch to a lazy initialization mode to save memory